### PR TITLE
Add warnings when trying to compute metrics for an unknown layer

### LINKED ISSUE #12068: Keras fails silently if the name of the output is not correct when using metrics.
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

```python
from keras import Model
from keras.layers import Input, Dense

input_tensor = Input((8,))
x = Dense(4)(input_tensor)
model = Model(input_tensor, x)

model.compile('sgd', 'mse', metrics={'random_inexistant_output_name': 'mae'})
# the previous line should throw an error listing the output names available and 
# printing the output name that  the user provided. It currently fails silently.
```

We should have an error message. Currently, the training works and the metric is just ignored.

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
### Summary

When giving unknown layer name in `metrics` to `model.compile`, will raise a warning instead of saying silent.

### Related Issues

fixes #12068 

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Can you also add the list of output names available in the warning?

**Comment by gabrieldemarmiesse** [comment]
And thanks for the quick PR!

**Comment by gabrieldemarmiesse** [comment]
Pro tip: put in your PR summary `fixes #{issue number} ` and the issue will be closed automatically when your PR is merged :)

File: keras/engine/training_utils.py [hunk 1]
```diff
@@ -322,6 +322,10 @@ def collect_metrics(metrics, output_names):
         return [copy.copy(metrics) for _ in output_names]
     elif isinstance(metrics, dict):
         nested_metrics = []
+        if not set(metrics.keys()).issubset(set(output_names)):
+            unknown_output_names = list(set(metrics.keys()) - set(output_names))
+            warnings.warn('Invalid layer name for metric computations: '
+                          '{}'.format(unknown_output_names))
```
Comment (gabrieldemarmiesse) [inline keras/engine/training_utils.py]: I would be faster for users to fix their code with the following message:

`Invalid output name for metric computation: 'incorrect_name_here', the output names of your model are 'output_names_here'`

**Comment by gabrieldemarmiesse** [comment]
Thanks a lot! This will make debugging much faster!
